File: test_db_connection.py
#!/usr/bin/env python3
"""
Script simples para testar a conexão com MySQL
"""
import os
from dotenv import load_dotenv

# Carrega variáveis de ambiente
load_dotenv()

try:
    import pymysql
    
    # Configurações do banco
    db_config = {
        'host': os.getenv('DB_HOST', '127.0.0.1'),
        'port': int(os.getenv('DB_PORT', 3306)),
        'user': os.getenv('DB_USER', 'root'),
        'password': os.getenv('DB_PASSWORD', 'Evo@000#!'),
        'database': os.getenv('DB_NAME', 'vibe'),
        'charset': 'utf8mb4'
    }
    
    print("🔍 Testando conexão com MySQL...")
    print(f"   Host: {db_config['host']}")
    print(f"   Port: {db_config['port']}")
    print(f"   User: {db_config['user']}")
    print(f"   Database: {db_config['database']}")
    
    # Tentar conectar
    connection = pymysql.connect(**db_config)
    print("✅ Conexão com MySQL estabelecida com sucesso!")
    
    # Testar uma query simples
    cursor = connection.cursor()
    cursor.execute("SELECT VERSION()")
    version = cursor.fetchone()
    print(f"🐬 Versão do MySQL: {version[0]}")
    
    # Verificar se existe o banco de dados
    cursor.execute("SHOW DATABASES")
    databases = [db[0] for db in cursor.fetchall()]
    
    if db_config['database'] in databases:
        print(f"✅ Banco de dados '{db_config['database']}' encontrado!")
    else:
        print(f"⚠️  Banco de dados '{db_config['database']}' não encontrado.")
        print("   Criando banco de dados...")
        cursor.execute(f"CREATE DATABASE IF NOT EXISTS {db_config['database']}")
        print(f"✅ Banco de dados '{db_config['database']}' criado!")
    
    cursor.close()
    connection.close()
    print("✅ Teste de conexão concluído com sucesso!")
    
except ImportError:
    print("❌ PyMySQL não está instalado. Execute: pip install pymysql")
except Exception as e:
    print(f"❌ Erro na conexão com MySQL: {e}")
    print("\n🔧 Verificações:")
    print("   1. MySQL está rodando?")
    print("   2. Credenciais estão corretas?")
    print("   3. Host e porta estão acessíveis?")

File: backend/email-service/test_db_connection.js
const mysql = require('mysql2/promise');
require('dotenv').config();

async function testDatabaseConnection() {
  console.log('🔍 Testando conexão com banco de dados...');
  
  const config = {
    host: process.env.DB_HOST || '127.0.0.1',
    port: process.env.DB_PORT || 3306,
    user: process.env.DB_USER || 'root',
    password: process.env.DB_PASSWORD || 'Evo@000#!',
    charset: 'utf8mb4'
  };
  
  console.log('📋 Configurações:');
  console.log(`   Host: ${config.host}`);
  console.log(`   Port: ${config.port}`);
  console.log(`   User: ${config.user}`);
  console.log(`   Password: ${'*'.repeat(config.password.length)}`);
  
  try {
    // Tentar conectar sem especificar banco
    console.log('\n🔄 Conectando ao MySQL...');
    const connection = await mysql.createConnection(config);
    
    console.log('✅ Conexão estabelecida!');
    
    // Verificar versão
    const [versionRows] = await connection.execute('SELECT VERSION() as version');
    console.log(`🐬 Versão do MySQL: ${versionRows[0].version}`);
    
    // Verificar usuário
    const [userRows] = await connection.execute('SELECT USER() as user');
    console.log(`👤 Usuário conectado: ${userRows[0].user}`);
    
    // Listar bancos
    const [dbRows] = await connection.execute('SHOW DATABASES');
    const databases = dbRows.map(row => Object.values(row)[0]);
    console.log(`📚 Bancos disponíveis: ${databases.join(', ')}`);
    
    // Criar banco 'vibe' se não existir
    if (!databases.includes('vibe')) {
      console.log('🔧 Criando banco de dados "vibe"...');
      await connection.execute('CREATE DATABASE vibe CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci');
      console.log('✅ Banco "vibe" criado!');
    } else {
      console.log('✅ Banco "vibe" já existe!');
    }
    
    // Testar conexão com o banco específico
    await connection.execute('USE vibe');
    console.log('✅ Conectado ao banco "vibe"!');
    
    await connection.end();
    
    console.log('\n🎉 Teste de conexão bem-sucedido!');
    return true;
    
  } catch (error) {
    console.error('\n❌ Erro na conexão:', error.message);
    
    if (error.code === 'ER_ACCESS_DENIED_ERROR') {
      console.log('\n🔧 Soluções para erro de acesso:');
      console.log('1. Verifique se a senha está correta');
      console.log('2. Execute no MySQL como administrador:');
      console.log('   ALTER USER \'root\'@\'localhost\' IDENTIFIED BY \'Evo@000#!\';');
      console.log('   FLUSH PRIVILEGES;');
      console.log('3. Ou use o arquivo fix_mysql_permissions.sql');
    } else if (error.code === 'ECONNREFUSED') {
      console.log('\n🔧 Soluções para erro de conexão:');
      console.log('1. Verifique se o MySQL está rodando');
      console.log('2. Verifique se a porta 3306 está aberta');
      console.log('3. Tente: net start mysql (Windows) ou sudo service mysql start (Linux)');
    }
    
    return false;
  }
}

// Executar teste se chamado diretamente
if (require.main === module) {
  testDatabaseConnection()
    .then(success => {
      process.exit(success ? 0 : 1);
    })
    .catch(error => {
      console.error('❌ Erro inesperado:', error);
      process.exit(1);
    });
}

module.exports = testDatabaseConnection;